ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala

```diff
@@ -44,133 +38,122 @@ private[auth] final class OngoingAuthorizationObserver[A](
   // 2) upstream component that is translating upstream Akka stream into [[onNext]] and other signals.
   private var afterCompletionOrError = false
 
-  @volatile private var lastUserRightsCheckTime = nowF()
+  private val cancelUserRightsChecks =
+    userRightsChecker.schedule(() => onError(staleStreamAuthError))
 
-  private lazy val userId = originalClaims.applicationId.fold[Ref.UserId](
-    throw new RuntimeException(
-      "Claims were resolved from a user but userId (applicationId) is missing in the claims."
-    )
-  )(Ref.UserId.assertFromString)
-
-  // Scheduling a task that periodically checks
-  // whether user rights state has changed.
-  // If user rights state has changed it aborts the stream by calling [[onError]]
-  private val cancellableO: Option[Cancellable] = {
-    if (originalClaims.resolvedFromUser) {
-      val delay = userRightsCheckIntervalInSeconds.seconds
-      // Note: https://doc.akka.io/docs/akka/2.6.13/scheduler.html states that:
-      // "All scheduled task will be executed when the ActorSystem is terminated, i.e. the task may execute before its timeout."
-      val c = akkaScheduler.scheduleWithFixedDelay(initialDelay = delay, delay = delay)(runnable =
-        checkUserRights _
-      )
-      Some(c)
-    } else None
-  }
-
-  private def checkUserRights(): Unit = {
-    userManagementStore
-      .listUserRights(userId)
-      .onComplete {
-        case Failure(_) | Success(Left(_)) =>
-          onError(staleStreamAuthError)
-        case Success(Right(userRights)) =>
-          val updatedClaims = AuthorizationInterceptor.convertUserRightsToClaims(userRights)
-          if (updatedClaims.toSet != originalClaims.claims.toSet) {
-            onError(staleStreamAuthError)
-          }
-          lastUserRightsCheckTime = nowF()
-      }
-  }
+  override def isCancelled: Boolean = synchronized(observer.isCancelled)
```

Comment:
switching to synchronized all over the place (to not have assumptions which part need and not need synchronization)

Reply:
These originally unsychronized members are, afaict from the doc strings, either safe to call from mutliple threads or they can be called only early in the lifecycle (i.e. ~ before the stream actually starts)